### PR TITLE
Add packet compression

### PR DESCRIPTION
**Pull request by nekiro**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Add missing packet compression for big packets

**Issues addressed:** 
None

**How to test:** 
Login to game, walk around, no crashes? works

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by nekiro** [comment]
> I was test this pr
> 
> ```
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_7UV282A0LJD6HKYT.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_WVPT28ZTJM2D5DRQ.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_PHA839T1F3WJNADU.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_UQQ0CKG94LMSIJTD.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_BY0BRRYLGSJPV7M2.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_CC66XSVYT629IPHD.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_O3UET4IWMWPEWHUH.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_JLP2YU5E50UTN58Y.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_3IXG4BE2WK51GMUG.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_W77GNRDQU6L8J5J6.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_DEI15LZQFC5MQXGQ.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_M5357L0YAAIQIYYX.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_CUC7PH7VGPESK21R.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_FZXZHF2F6A11DIV8.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_6PQ6CCA4WWXC3I24.cpp')
> 1>D:\src\outputmessage.cpp(25,55): warning C4459: declaration of 'bufferedProtocols' hides global declaration
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_CUC7PH7VGPESK21R.cpp')
> 1>D:\src\outputmessage.cpp(21,27):
> 1>see declaration of '`anonymous-namespace'::bufferedProtocols'
> 1>D:\src\outputmessage.cpp(31,47): warning C4459: declaration of 'bufferedProtocols' hides global declaration
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_CUC7PH7VGPESK21R.cpp')
> 1>D:\src\outputmessage.cpp(21,27):
> 1>see declaration of '`anonymous-namespace'::bufferedProtocols'
> 1>unity_8NZJOI7GO26MNSLA.cpp
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_FZXZHF2F6A11DIV8.cpp')
> 1>D:\src\protocol.h(55,22): warning C4018: '>=': signed/unsigned mismatch
> 1>(compiling source file '/theforgo.A10F9657/x64/Release/unity_8NZJOI7GO26MNSLA.cpp')
> ```
> 
> in cipsoft client works fine
> 
> but in otc by enabling the feature "GameSequencedPackets" `g_game.enableFeature(GameSequencedPackets)`
> 
> [mehah/otclient@4e79004](https://github.com/mehah/otclient/commit/4e79004b891a1ca494b853651e3cad20d13ca53b) "SPX Protocol Support (with compress)"
> 
> error
> 
> ```
> ERROR: failed to decompress message - too many length or distance symbols
> at:
>     [C++]: ?internalRecvData@Protocol@@AEAAXPEAEG@Z
>     
>     ERROR: failed to decompress message - invalid stored block lengths
> at:
>     [C++]: ?internalRecvData@Protocol@@AEAAXPEAEG@Z
> ```

If cip client works, then it must be something different at otclient side.

**Comment by kokekanon** [comment]
> > in cipsoft client works fine
> > but in otc by enabling the feature "GameSequencedPackets" `g_game.enableFeature(GameSequencedPackets)`
> > [mehah/otclient@4e79004](https://github.com/mehah/otclient/commit/4e79004b891a1ca494b853651e3cad20d13ca53b) "SPX Protocol Support (with compress)"
> 
> If cip client works, then it must be something different at otclient side or incorrect configuration.

the client is fine, the problem is tfs, this if does not integrate OTCR windows.

```lua
	if (version >= 1111 && operatingSystem >= CLIENTOS_QT_LINUX && operatingSystem < CLIENTOS_OTCLIENT_LINUX) {
		setChecksumMode(CHECKSUM_SEQUENCE);
	}
```
image
I confirm that it works in OTCR

player open higscore module

**Server** 
packet size before: 4360
packet size after: 1734

**Client**

WARNING: raw: 4360
WARNING: compression: 1734

**Comment by nekiro** [comment]
I assume change works then on both clients.

**Comment by kojimhojo** [comment]
thanks. 
i test it and works 
ty

**Review by kojimhojo — APPROVED** [review]

**Comment by ArturKnopik** [comment]
Tested on Cipsoft client, it's working.

File: CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,7 @@ endif ()
 
 # Find packages.
 find_package(OpenSSL 3.0.0 REQUIRED COMPONENTS Crypto)
+find_package(ZLIB REQUIRED)
```
Comment (ArturKnopik) [inline CMakeLists.txt:52]: Require update in wiki about new zlib dependency.

**Review by ArturKnopik — APPROVED** [review]

**Pull request merged**